R/flatten_data.R

```diff
@@ -1,7 +1,42 @@
+#' determine data type
+#' 
+#' @param data (list or tbl_df, tbl, data.frame) An ecocomDP formatted dataset (list), or a list of ecocomDP tables, a single flat ecocomDP table, or a single flat table created by joinng and spreading ecocomDP tables. 
+#' 
+#' @return (character) A character string indicating data type. One of: 'list_of_tables', 'datset_old', 
+#' 'dataset', or 'flat_table'
+#' 
+#' @details Internal function for detecting datatype. Not exported
+detect_data_type <- function(data){
+  
+  table_names <- c(
+    "location",
+    "location_ancillary",
+    "taxon",
+    "observation",
+    "observation_ancillary",
+    "dataset_summary")
```

Comment:
@sokole Add the variable_mapping table to this list.

Reply:
done